src/Microsoft.ML.Data/Scorers/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,487 @@
+                }
+                else
+                {
+                    _outputSchema = new WtfSchema(_env, DefaultColumnNames.FeatureContributions,
+                        new VectorType(NumberType.R4, schema.Feature.Type.AsVector),
+                        InputSchema, InputRoleMappedSchema.Feature.Index);
+                }
+
+                _outputGenericSchema = _genericRowMapper.Schema;
+                Schema = new CompositeSchema(new ISchema[] { _outputGenericSchema, _outputSchema, }).AsSchema;
+            }
+
+            /// <summary>
+            /// Returns the input columns needed for the requested output columns.
+            /// </summary>
+            public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+            {
+                for (int i = 0; i < Schema.ColumnCount; i++)
+                {
+                    if (predicate(i))
+                        return col => col == InputRoleMappedSchema.Feature.Index;
+                }
+                return col => false;
+            }
+
+            public IRow GetOutputRow(IRow input, Func<int, bool> predicate, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.AssertValue(predicate);
+                var totalColumnsCount = 1 + _outputGenericSchema.ColumnCount;
+                var getters = new Delegate[totalColumnsCount];
+
+                if (predicate(totalColumnsCount - 1))
+                {
+                    getters[totalColumnsCount - 1] = _parent.Stringify
+                        ? _parent.GetTextWtfGetter(input, InputRoleMappedSchema.Feature.Index, _slotNames)
+                        : _parent.GetWtfGetter(input, InputRoleMappedSchema.Feature.Index);
+                }
+
+                var genericRow = _genericRowMapper.GetRow(input, GetGenericPredicate(predicate), out disposer);
+                for (var i = 0; i < _outputGenericSchema.ColumnCount; i++)
+                {
+                    if (genericRow.IsColumnActive(i))
+                        getters[i] = RowCursorUtils.GetGetterAsDelegate(genericRow, i);
+                }
+
+                return new SimpleRow(Schema, input, getters);
+            }
+
+            public Func<int, bool> GetGenericPredicate(Func<int, bool> predicate)
+            {
+                return col => predicate(col);
+            }
+
+            public IEnumerable<KeyValuePair<RoleMappedSchema.ColumnRole, string>> GetInputColumnRoles()
+            {
+                yield return RoleMappedSchema.ColumnRole.Feature.Bind(InputRoleMappedSchema.Feature.Name);
+            }
+
+            public IRow GetRow(IRow input, Func<int, bool> active, out Action disposer)
+            {
+                return GetOutputRow(input, active, out disposer);
+            }
+        }
+
+        private sealed class WtfSchema : ISchema
```

Comment:
>Wtf [](start = 29, length = 3)

I would suggest to search "Wtf" in this code, and replace it with something else.

Reply:
I suggest `FeatureContribution`.